catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternFormulaUtils.java

```diff
@@ -591,7 +609,17 @@ public static boolean applyBracketCorrection(List<InternToken> internFormula) th
 					break;
 			}
 		}
-		return true;
+		if (stack.isEmpty()) {
+			return true;
+		}
+		if (stack.peek() == InternTokenType.BRACKET_OPEN) {
+			InternFormula.stackStatusBrackets =
+					InternFormula.StackStatusBrackets.BRACKET_OPEN_STACK_NOT_EMPTY;
+		} else if (stack.peek() == InternTokenType.FUNCTION_PARAMETERS_BRACKET_OPEN) {
+			InternFormula.stackStatusBrackets =
+					InternFormula.StackStatusBrackets.FUNCTION_PARAMETERS_BRACKET_OPEN_STACK_NOT_EMPTY;
+		}
+		return stack.empty();
```

Comment:
You are using once a vector method `stack.isEmpty()` (line 612) and once a stack method `stack.empty()` (line 622) on the same stack variable, is there any specific reason for that? Otherwise please choose one and stick to that method when you want to check if the stack is empty.

Reply:
No there was no reason for that, i changed it now. Thanks for the review!